scripts/py_matter_yamltests/matter_yamltests/constraints.py

```diff
@@ -42,7 +37,13 @@ def is_met(self, value):
             return self._is_null_allowed
 
         response_type = type(value)
-        if self._types and response_type not in self._types:
+        type_matches = False if self._types else True
```

Comment:
This feels a bit off as a pattern like `x = True if y else False` which seems like `x = y` (except it is a bool cast).

Not sure if more readable, but maybe:

```py
type_matches = not self._types or any([
    issubclass(response_type, expected) for expected in self._types
])
```